content/courses/advanced-ada/parts/data_types/numerics.rst

```diff
@@ -1329,22 +1329,457 @@ input parameter :ada:`V` (of :ada:`One_To_Ten'Base` type) in the runtime output
+    standard :ada:`Integer` type:
+
+    .. code:: ada run_button project=Courses.Advanced_Ada.Data_Types.Numerics.Discrete_Numeric_Types.GNAT_Integer_Types
+
+        with Ada.Text_IO; use Ada.Text_IO;
+
+        procedure Show_GNAT_Integer_Types is
+        begin
+
+           Put_Line ("Short_Short_Integer'Size:    "
+                     & Short_Short_Integer'Size'Image
+                     & " bits");
+
+           Put_Line ("Short_Integer'Size:          "
+                     & Short_Integer'Size'Image
+                     & " bits");
+
+           Put_Line ("Integer'Size:                "
+                     & Integer'Size'Image
+                     & " bits");
+
+           Put_Line ("Long_Integer'Size:           "
+                     & Long_Integer'Size'Image
+                     & " bits");
+
+           Put_Line ("Long_Long_Integer'Size:      "
+                     & Long_Long_Integer'Size'Image
+                     & " bits");
+
+           Put_Line ("Long_Long_Long_Integer'Size: "
+                     & Long_Long_Long_Integer'Size'Image
+                     & " bits");
+
+        end Show_GNAT_Integer_Types;
+
+    The actual range of each of these integer types depends on the target
+    architecture. (Note that you may have different types with the same range.)
+
+    Also, when interfacing with C code, GNAT guarantees the following type
+    equivalence:
+
+    +------------------------------+------------------------------------------+
+    | C type                       | Ada type                                 |
+    +==============================+==========================================+
+    | :c:`signed char`             | :ada:`Short_Short_Integer`               |
+    +------------------------------+------------------------------------------+
+    | :c:`short int`               | :ada:`Short_Integer`                     |
+    +------------------------------+------------------------------------------+
+    | :c:`int`                     | :ada:`Integer`                           |
+    +------------------------------+------------------------------------------+
+    | :c:`long`                    | :ada:`Long_Integer`                      |
+    +------------------------------+------------------------------------------+
+    | :c:`long long`               | :ada:`Long_Long_Integer`                 |
+    +------------------------------+------------------------------------------+
+
+
+.. _Adv_Ada_Custom_Integer_Types:
+
+Custom integer types
+^^^^^^^^^^^^^^^^^^^^
+
+As we've mentioned before, numeric data types such as :ada:`Integer` or
+:ada:`Long_Integer` might not provide enough control over the ranges |mdash|
+i.e. the range selected by the compiler may not correspond to the expected
+range of the numeric algorithm we're implementing. Therefore, it is best to
+simply define custom ranges. To do that, you should evaluate the algorithm and
```

Comment:
I think you could simplify the above to 
"As we've mentioned before, for the language-defined numeric data types such as :ada:`Integer` or
:ada:`Long_Integer` the range selected by the compiler may not correspond to the required
range of the numeric algorithm we're implementing. Therefore, it is best to
simply declare custom types with the necessary ranges specified."